### PR TITLE
Bugfix/KITOSUDV-4775 fix FK organization query time

### PR DESCRIPTION
**Pull request by a-naskret**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [ ] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [ ] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [ ] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [ ] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Core.ApplicationServices/Organizations/Handlers/AuthorizedUpdateOrganizationFromFKOrganisationCommandHandler.cs
Comment (strongmindsmhs) [inline Core.ApplicationServices/Organizations/Handlers/AuthorizedUpdateOrganizationFromFKOrganisationCommandHandler.cs:1]: These are only small changes, I know, but, the Execute method could at some point thrive with refactorings to smaller private methods

Comment (mojtrifork) [reply Core.ApplicationServices/Organizations/Handlers/AuthorizedUpdateOrganizationFromFKOrganisationCommandHandler.cs:1]: Agreed and the same goes for the big service which is as it was on organization - just pointing to the orgsystem. I would argue though to wait with that refactoring until a later data - to not introduce too much risk :-)

File: Infrastructure.STS.OrganizationSystem/DomainServices/StsOrganizationSystemService.cs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            _organizationService = organizationService;
+            _logger = logger;
+            _certificateThumbprint = configuration.CertificateThumbprint;
+            _serviceRoot = $"https://{configuration.EndpointHost}/service/Organisation/OrganisationSystem/5";
+        }
+
+        public Result<ExternalOrganizationUnit, DetailedOperationError<ResolveOrganizationTreeError>> ResolveOrganizationTree(Organization organization)
+        {
+            //Search for org units by org uuid
+            using var clientCertificate = X509CertificateClientCertificateFactory.GetClientCertificate(_certificateThumbprint);
```
Comment (mojtrifork) [inline Infrastructure.STS.OrganizationSystem/DomainServices/StsOrganizationSystemService.cs:37]: **This is a big beast** but to not introduce too much risk (I know much of this is carried over from the old service) we should not retain it as is. It surprises me just how compatible the code was

File: Infrastructure.STS.OrganizationSystem/DomainServices/StsOrganizationSystemService.cs
Comment (strongmindsmhs) [inline Infrastructure.STS.OrganizationSystem/DomainServices/StsOrganizationSystemService.cs:1]: Yes, also a candidate for chopping up the long method .. if not now, then park the needed work in a technical dept Jira

Comment (mojtrifork) [reply Infrastructure.STS.OrganizationSystem/DomainServices/StsOrganizationSystemService.cs:1]: agreed - we add a tech story for it... but then again we will rewrite all of the integrations early 2024, so might just be a reminder not to redo the issue

**Review by strongmindsmhs — APPROVED** [review]
In general: Are all the xsd schemas neccesary for referencing just the one search method?

**Pull request merged**